src/ape/managers/chain.py

```diff
@@ -1645,4 +1645,50 @@ def set_balance(self, account: Union[BaseAddress, AddressType], amount: Union[in
         return self.provider.set_balance(account, amount)
 
     def get_receipt(self, transaction_hash: str) -> ReceiptAPI:
+        """
+        Get a transaction receipt from the chain.
+
+        Args:
+            transaction_hash (str): The hash of the transaction.
+
+        Returns:
+            :class:`~ape.apt.transactions.ReceiptAPI`
+        """
         return self.chain_manager.history[transaction_hash]
+
+    def get_contract_receipt(
```

Comment:
Not a fan of this name or location, could it be `ContractCache.get_creation_receipt(addr)`?

Reply:
Yes, I will make this change!

But I am not 100% convinced `ContractCache` is actually the right place for this because of the frequent use of `BlockContainer` is this logic. That hints to me that this belongs at a higher level than `ContractCache` or something.

Reply:
it uses `ProviderAPI.get_code`, not sure where it is more logically situated to better take advantage of the block container (except current chain head), but semantically speaking it's directly related to contracts (we also cache it in the contract)